pkg/server/server.go

```diff
@@ -568,13 +570,13 @@ func (s *Server) Close() {
 }
 
 func (s *Server) ListObjects(ctx context.Context, req *openfgav1.ListObjectsRequest) (*openfgav1.ListObjectsResponse, error) {
+	start := time.Now()
```

Comment:
Let's put this in a middleware instead 😄 

In https://github.com/openfga/openfga/blob/57dd34cd538b487ef162694c5d36384cf2fb6f38/pkg/middleware/logging/logging.go#L58-L59 you can add this:

```go
func (r *reporter) PostCall(err error, _ time.Duration) {
	duration := time.Since(r.start)
	r.transport.SetHeader(r.ctx, "fga-query-duration-ms", fmt.Sprintf("%d", duration.Milliseconds()))
	r.fields = append(r.fields, zap.Duration("took", duration))
```

which will both send a header and write to the log.

You'll have to edit the constructors of this middleware to pass a `transport` field that you can get from here: https://github.com/openfga/openfga/blob/0f960edd0b9455340a835241396c504c5d96871d/cmd/run/run.go#L587

Also you'll have to update these tests: `TestHTTPHeaders` and `TestServerLogs`.